django/http/request.py

```diff
@@ -131,7 +143,7 @@ def get_preferred_type(self, media_types):
             return None
 
         # Of the desired media types, select the one which is most desirable.
-        return min(desired_types, key=lambda t: self.accepted_types.index(t[0]))[1]
+        return max(desired_types, key=lambda t: t[0].quality)[1]
```

Comment:
When I revert this change the tests still pass, but I _think_ that's because these are basically logically equivalent (due to the way `self.accepted_types` is ordered) and that this is easier to read

I think I would like a test for `get_preferred_type` which was failing before and passing after this commit (as `get_preferred_type` was the focus of the ticket)